### ISSUE TITLE
Hidden input triggers "`value` prop without an `onChange` handler" prop type error

### ISSUE THREAD

**Issue by kkkrist** [sanusart/react-dropdown-select#74]
I just stumbled upon this little problem: If you set the `name` or `required` prop, the hidden input at [`src/index.js#L469`](https://github.com/sanusart/react-dropdown-select/blob/master/src/index.js#L469) will trigger the following error:

```
Warning: Failed prop type: You provided a `value` prop to a form field without an `onChange` handler. This will render a read-only field. If the field should be mutable use `defaultValue`. Otherwise, set either `onChange` or `readOnly`.
```

While this is technically a read-only input, setting `readOnly` would make the form ignore the `required` attribute. So I guess in order to make this error go away properly, the hidden input should either look like this:

```javascript
<input
  defaultValue={this.state.values.map(value => value[this.props.labelField]).toString() || []}
  disabled={this.props.disabled}
  key={this.state.values}
  name={this.props.name}
  pattern={this.props.pattern}
  required={this.props.required}
  style={{ opacity: 0, width: 0, position: 'absolute' }}
  tabIndex={-1}
/>
```

or this (if we really need `value`):

```javascript
<input
  disabled={this.props.disabled}
  name={this.props.name}
  onChange={() => {}}
  pattern={this.props.pattern}
  required={this.props.required}
  style={{ opacity: 0, width: 0, position: 'absolute' }}
  tabIndex={-1}
  value={this.state.values.map(value => value[this.props.labelField]).toString() || []}
/>
```

What do you think?

**Comment by sanusart** [comment]
@kkkrist Yes, sounds reasonable.

**Comment by kkkrist** [comment]
Should I do some testing and do a PR? Which variant do you prefer?

**Comment by sanusart** [comment]
Sure if you're up to it could be great.
When you open a pr it triggers a build in codesandbox we can test there.
I'd preferere the defaultValue one, seems cleaner and without empty
onChange.

**Comment by kkkrist** [comment]
Looks like the `key` hack isn't necessary after all, nice. Also fixed the initial value for `pattern` along the way. All warnings gone now in my `<form>` use case.

**Comment by sanusart** [comment]
Thank you @kkkrist, looks great, published version: 4.0.1
